src/features/perps/components/AmountInputCard/AmountInputCard.tsx

```diff
@@ -364,13 +386,19 @@ export const AmountInputCard = memo(function AmountInputCard() {
     [balanceValue, displayedAmount, inputSource, lastAccountAddress, resetToInitial, sliderProgress]
   );
 
-  useListen(
-    useHlNewPositionStore,
-    state => state.marketResetSignal,
-    () => resetToInitial()
-  );
+  useEffect(() => {
+    resetToInitial(availableBalance);
+  }, [availableBalance, resetKey, resetToInitial]);
+
+  useEffect(() => {
+    revalidateAmount(availableBalance);
+  }, [availableBalance, revalidateAmount]);
```

Comment:
Did you check that this doesn't cause issues with the initial state?

This changes the logic slightly, as `useListen` only fires on change by default, not on mount like `useEffect`.

Might make sense to use `useOnChange` which is just `useEffect` minus the initial call.